datastore/elastic_search/query.py

```diff
@@ -235,7 +287,72 @@ def _generate_es_ngram_search_dictionary(entity_name, ngrams_list, fuzziness_thr
         },
         'number_of_fragments': 20
     }
+    return data
 
+
+def _generate_es_search_dictionary(entity_name, user_typed_text, fuzziness_threshold, language_script=None):
+    """
+    Generates compound elasticsearch boolean search query dictionary for the user says sentence. The query generated
+    searches for entity_name in the index and returns search results for the matched word (of user-says sentence)
+     only if entity_name is found.
+
+    Args:
+        entity_name: name of the entity to perform a 'term' query on
+        user_typed_text: The sentence typed by user on which we need to identify the enitites.
+        fuzziness_threshold: fuzziness_threshold for elasticsearch match query 'fuzziness' parameter
+        language_script: language of documents to be searched, optional, defaults to None
+
+    Returns:
+        dictionary, the search query for the user typed sentence
+
+    """
+    must_terms = []
+    term_dict_entity_name = {
+        'term': {
+            'entity_data': {
+                'value': entity_name
+            }
+        }
+    }
+    must_terms.append(term_dict_entity_name)
+
+    if language_script is not None:
+        term_dict_language = {
+            'term': {
+                'language_script': {
+                    'value': language_script
+                }
+            }
+        }
+        must_terms.append(term_dict_language)
+
+    data = {
+        'query': {
+            'bool': {
+                'must': must_terms,
+                'should': [],
+                'minimum_number_should_match': 1
+            }
+        }
+    }
+    query_should_data = []
+    query = {
+        'match': {
+            'variants': {
+                'query': user_typed_text,
+                'fuzziness': _get_dynamic_fuzziness_threshold(user_typed_text, fuzziness_threshold),
```

Comment:
This will cause problems for elasticsearch < 6.2 or when a constant fuzziness is set. Fuzziness, when applied to short words with length less than 3, has potential to increase false positives.

The method `_get_dynamic_fuzziness_threshold` was designed to work with words, not sentences! It will return 2 in most cases if fuzziness is set to `autox:y` mode